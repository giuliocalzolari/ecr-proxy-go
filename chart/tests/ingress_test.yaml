suite: test ingress
templates:
- ingress.yaml
release:
  name: test-release
  namespace: test-namespace
tests:
- it: manifest should match snapshot
  chart:
    version: 1.2.3
  set:
    ingress:
      enabled: true
  asserts:
  - matchSnapshot: {}




- it: should add annotations
  set:
    ingress:
      enabled: true
      annotations:
        foo: bar
  asserts:
  - equal:
      path: metadata.annotations.foo
      value: bar


- it: should set tls if given
  set:
    ingress:
      enabled: true
      tls:  true
      existingSecret: my-tls-secret
      path: /test
      pathType: ImplementationSpecific
  asserts:
  - equal:
      path: spec.tls
      value:
      - hosts:
        - chart-example.local
        secretName: my-tls-secret
  - equal:
      path: spec.rules[0].http.paths[0].path
      value: /test
  - equal:
      path: spec.rules[0].http.paths[0].pathType
      value: ImplementationSpecific



- it: should set ingressClassName
  set:
    ingress:
      enabled: true
      ingressClassName: test
  asserts:
  - equal:
      path: spec.ingressClassName
      value: test

