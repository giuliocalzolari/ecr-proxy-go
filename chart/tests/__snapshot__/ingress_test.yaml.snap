manifest should match snapshot:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: ecr-proxy
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: ecr-proxy-1.2.3
      name: ecr-proxy
    spec:
      ingressClassName: nginx
      rules:
        - host: chart-example.local
          http:
            paths:
              - backend:
                  service:
                    name: ecr-proxy
                    port:
                      number: 5000
                path: /
                pathType: Prefix
      tls:
        - hosts:
            - chart-example.local
          secretName: chart-example.local-tls
